### DISCUSSION TITLE
Problems with User Access

### DISCUSSION THREAD

**Discussion by GuilhermeHirsch** [Budibase/budibase#1992] [Help] [answered]
Hey guys!
I created a second user for testing.
This user is not getting access to the app, a white screen appears after login (successfully). He gets access only when he has developer permissions.
The user is configured (in the app and globally) as admin, and all screens have "basic" permission. Where is my mistake?

**Comment by joebudi** [comment, marked as answer]
Hey @GuilhermeHirsch 

Are there any errors appearing in the console when the user logs in?

**Reply by GuilhermeHirsch** [reply]
Helo, @joebudi! Apparently no error is presented: https://www.loom.com/share/2f7377888de94dc9af6d04e7a41921cf

**Reply by joebudi** [reply]
Thanks for the Loom!

If you can try:
-  right-clicking the white screen
- select Inspect
- and locate the red error button (image below)
![Screenshot 2021-07-14 at 16 55 26](https://user-images.githubusercontent.com/49767913/125653260-a2bac83d-5de7-4611-9754-a53faaf93ff4.png)
- click the red button
- and grab a screenshot of the console log (see the red writing/error message in the image below as an example)
![Screenshot 2021-07-14 at 16 56 16](https://user-images.githubusercontent.com/49767913/125653383-09f098db-3475-4310-819b-d6ba3c441ba2.png)
- and add the screenshot to this discussion 👍

**Reply by GuilhermeHirsch** [reply]
![Erro](https://user-images.githubusercontent.com/33108596/125655103-e2f6e086-bf0f-47ee-ac22-d58b17087612.jpg)

**Reply by joebudi** [reply]
Thanks @GuilhermeHirsch . I will look into this now and respond asap.

**Reply by mike12345567** [reply]
I @GuilhermeHirsch could you tell me what your user panel looks like? e.g. what development/admin access has been granted to the user?

Like below all of my test users have been granted both development and admin access.
![image](https://user-images.githubusercontent.com/4407001/125655812-b3cd9bd8-54cb-44b9-8791-497c0c94c784.png)

**Reply by GuilhermeHirsch** [reply]
It's with the first user I can't access (andrehirsch@viavale.com.br).

![Erro 2](https://user-images.githubusercontent.com/33108596/125656300-b9e3f9f8-ddd5-41e9-92ea-93630d511b6f.jpg)

**Reply by mike12345567** [reply]
Hi @GuilhermeHirsch - I realise this is a bit confusing, but the "admin access" really only grants the user the ability to administrate the Budibase platform through the portal, it doesn't grant any access to apps.

If you grant the first user development access they will be able to view every app in the platform automatically, or if you doing any them to be able to build apps you'll need to go into the user and grant them basic access to the app.

Edit: there is also some problems in the currently released version, when you only have a single app it skips the portal stage and jumps straight into the app, but in this case your user doesn't have access to the app so they get rejected. We have a wide variety of fixes for things around this coming soon, as well as making it more clear to the user when they don't have permission to view something.

**Reply by GuilhermeHirsch** [reply]
This video maybe better contextualizes:

https://www.loom.com/share/c502e6c1b8ac4a09a1f870c70a9a4b8c

**Reply by mike12345567** [reply]
Hi @GuilhermeHirsch - could you try revoking admin access from your user and then sending me a picture of the screen once you've clicked into that user? Like below:

![image](https://user-images.githubusercontent.com/4407001/125658956-e050c567-2c77-4589-a41a-147240bbb5db.png)

Not 100% sure whats going on, but I think its related to the fact they only have admin access, realistically we shouldn't have allowed this configuration as right now it doesn't really do anything (other than possibly break things!)

**Reply by GuilhermeHirsch** [reply]
![Erro 3](https://user-images.githubusercontent.com/33108596/125660487-b15e9cb0-2cd8-47e5-a069-ce4abd2baeca.jpg)

**Reply by mike12345567** [reply]
Interesting, it looks like your list of apps is empty, that section where you configure roles below is where you are supposed to be able to configure the roles for your user.

I've been trying to re-create your issue, to no success, is it possible you could re-create the user, but not enable their administration access - instead once the user is created go into this page and configure their roles for the apps?

**Reply by GuilhermeHirsch** [reply]
Now I've noticed that my main user is behaving the same:

![Erro 4](https://user-images.githubusercontent.com/33108596/125662707-2da85ace-5c01-4c40-978f-7c0b420bf861.jpg)

**Reply by mike12345567** [reply]
Hmm this is very odd, could you check for me if you're up to date with the latest version? I know theres been a ton of fixes in this area over the last few weeks.

**Reply by GuilhermeHirsch** [reply]
Current Version: 0.9.59

**Reply by mike12345567** [reply]
ah I think this might be the problem, could you possibly update? Theres been a few bug fixes released since that version (I think we're on 0.9.74 now).

**Reply by GuilhermeHirsch** [reply]
I created another user for testing (basic onboarding) without admin and developer access.
After that I created a new permission rule in the app and assigned it to a new user. Even so the "Configure roles" table is empty.

**Reply by GuilhermeHirsch** [reply]
I have it in a docker in the digital ocean. What is the command to update on the console? I never updated and I don't want to lose my app.

**Reply by GuilhermeHirsch** [reply]
Updated

**Reply by GuilhermeHirsch** [reply]
After the update I got 🙏:

![Erro 5](https://user-images.githubusercontent.com/33108596/125667711-22b2ecf6-e151-4626-aae5-2a34857c579c.jpg)

I was basing myself on the GitHub version:

![Sem título](https://user-images.githubusercontent.com/33108596/125668225-6d3e751a-fda0-4ee0-8b4b-6ec9ba91ce9d.jpg)

**Reply by GuilhermeHirsch** [reply]
Thank you so much for your help, @joebudi and @mike12345567!

**Reply by mike12345567** [reply]
Sorry @GuilhermeHirsch was away for a bit there, glad you got it sorted! Ah yes, that is a little bit confusing, we plan to make the process of updating a lot easier as part of our future work, making it possible to check/update within app.
